// @trycourier/react-elements/dist/components/color-pickers/medium.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.MediumColorPicker = void 0;

var _react = _interopRequireDefault(require("react"));

var _styledComponents = _interopRequireDefault(require("styled-components"));

var _description = require("../description");

var MediumColorPicker = function MediumColorPicker(_ref) {
  var value = _ref.value,
      label = _ref.label,
      _onChange = _ref.onChange;
  return /*#__PURE__*/_react["default"].createElement(ColorPickerContainer, null, /*#__PURE__*/_react["default"].createElement(ColorPreview, {
    color: value
  }), /*#__PURE__*/_react["default"].createElement(_description.Description, null, label), /*#__PURE__*/_react["default"].createElement(ColorHex, {
    color: value
  }, value.toUpperCase()), /*#__PURE__*/_react["default"].createElement("input", {
    type: "color",
    name: label,
    value: value,
    style: {
      display: "none"
    },
    onChange: function onChange(e) {
      return _onChange(e.target.value);
    }
  }));
};

exports.MediumColorPicker = MediumColorPicker;

var ColorPickerContainer = _styledComponents["default"].label.withConfig({
  displayName: "medium__ColorPickerContainer",
  componentId: "sc-14fitb2-0"
})(function () {
  return {
    "&:hover": {
      cursor: "pointer"
    }
  };
});

var ColorPreview = _styledComponents["default"].div.withConfig({
  displayName: "medium__ColorPreview",
  componentId: "sc-14fitb2-1"
})(function (_ref2) {
  var color = _ref2.color;
  return {
    backgroundColor: color,
    borderRadius: "100%",
    width: "35px",
    height: "35px"
  };
});

var ColorHex = _styledComponents["default"].p.withConfig({
  displayName: "medium__ColorHex",
  componentId: "sc-14fitb2-2"
})(function (_ref3) {
  var color = _ref3.color;
  return {
    fontFamily: "'Nunito Sans', sans-serif",
    fontWeight: 600,
    fontStyle: "normal",
    fontSize: "12px",
    lineHeight: "16px",
    margin: "0",
    color: color
  };
});
