// @trycourier/react-toast/dist/components/Toast/styled.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.ToastStyled = exports.toastStyles = void 0;

var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));

var _styledComponents = _interopRequireDefault(require("styled-components"));

var _reactToastify = require("react-toastify");

function ownKeys(object, enumerableOnly) { var keys = Object.keys(object); if (Object.getOwnPropertySymbols) { var symbols = Object.getOwnPropertySymbols(object); if (enumerableOnly) symbols = symbols.filter(function (sym) { return Object.getOwnPropertyDescriptor(object, sym).enumerable; }); keys.push.apply(keys, symbols); } return keys; }

function _objectSpread(target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i] != null ? arguments[i] : {}; if (i % 2) { ownKeys(Object(source), true).forEach(function (key) { (0, _defineProperty2["default"])(target, key, source[key]); }); } else if (Object.getOwnPropertyDescriptors) { Object.defineProperties(target, Object.getOwnPropertyDescriptors(source)); } else { ownKeys(Object(source)).forEach(function (key) { Object.defineProperty(target, key, Object.getOwnPropertyDescriptor(source, key)); }); } } return target; }

var toastStyles = function toastStyles(_ref) {
  var _theme$brand$inapp$to, _theme$brand, _theme$brand$inapp, _theme$brand$inapp$to2, _theme$brand$inapp$to3, _theme$brand2, _theme$brand2$inapp, _theme$brand2$inapp$t, _theme$brand$colors$p, _theme$brand3, _theme$brand3$colors, _theme$brand$inapp$to4, _theme$brand4, _theme$brand4$inapp, _theme$brand4$inapp$t, _theme$brand$inapp$to5, _theme$brand5, _theme$brand5$inapp, _theme$brand5$inapp$t, _ref2;

  var theme = _ref.theme;
  return _ref2 = {}, (0, _defineProperty2["default"])(_ref2, "&.Toastify__toast-container", _objectSpread({
    "&, *": {
      fontFamily: "'Nunito Sans', sans-serif",
      boxSizing: "border-box"
    }
  }, theme === null || theme === void 0 ? void 0 : theme.root)), (0, _defineProperty2["default"])(_ref2, ".Toastify__toast", _objectSpread({
    width: 320,
    minHeight: 65,
    padding: 0,
    borderRadius: "6px",
    borderBottomLeftRadius: (_theme$brand$inapp$to = theme === null || theme === void 0 ? void 0 : (_theme$brand = theme.brand) === null || _theme$brand === void 0 ? void 0 : (_theme$brand$inapp = _theme$brand.inapp) === null || _theme$brand$inapp === void 0 ? void 0 : (_theme$brand$inapp$to2 = _theme$brand$inapp.toast) === null || _theme$brand$inapp$to2 === void 0 ? void 0 : _theme$brand$inapp$to2.borderRadius) !== null && _theme$brand$inapp$to !== void 0 ? _theme$brand$inapp$to : "6px",
    borderBottomRightRadius: (_theme$brand$inapp$to3 = theme === null || theme === void 0 ? void 0 : (_theme$brand2 = theme.brand) === null || _theme$brand2 === void 0 ? void 0 : (_theme$brand2$inapp = _theme$brand2.inapp) === null || _theme$brand2$inapp === void 0 ? void 0 : (_theme$brand2$inapp$t = _theme$brand2$inapp.toast) === null || _theme$brand2$inapp$t === void 0 ? void 0 : _theme$brand2$inapp$t.borderRadius) !== null && _theme$brand$inapp$to3 !== void 0 ? _theme$brand$inapp$to3 : "6px",
    overflow: "visible"
  }, theme === null || theme === void 0 ? void 0 : theme.toast)), (0, _defineProperty2["default"])(_ref2, ".Toastify__toast-body", _objectSpread({
    margin: 0,
    display: "flex",
    justifyContent: "space-between"
  }, theme === null || theme === void 0 ? void 0 : theme.body)), (0, _defineProperty2["default"])(_ref2, ".Toastify__progress-bar", _objectSpread({
    background: (_theme$brand$colors$p = theme === null || theme === void 0 ? void 0 : (_theme$brand3 = theme.brand) === null || _theme$brand3 === void 0 ? void 0 : (_theme$brand3$colors = _theme$brand3.colors) === null || _theme$brand3$colors === void 0 ? void 0 : _theme$brand3$colors.primary) !== null && _theme$brand$colors$p !== void 0 ? _theme$brand$colors$p : "#9121C2",
    height: 3,
    top: 0,
    zIndex: 25,
    borderTopLeftRadius: (_theme$brand$inapp$to4 = theme === null || theme === void 0 ? void 0 : (_theme$brand4 = theme.brand) === null || _theme$brand4 === void 0 ? void 0 : (_theme$brand4$inapp = _theme$brand4.inapp) === null || _theme$brand4$inapp === void 0 ? void 0 : (_theme$brand4$inapp$t = _theme$brand4$inapp.toast) === null || _theme$brand4$inapp$t === void 0 ? void 0 : _theme$brand4$inapp$t.borderRadius) !== null && _theme$brand$inapp$to4 !== void 0 ? _theme$brand$inapp$to4 : "6px",
    borderTopRightRadius: (_theme$brand$inapp$to5 = theme === null || theme === void 0 ? void 0 : (_theme$brand5 = theme.brand) === null || _theme$brand5 === void 0 ? void 0 : (_theme$brand5$inapp = _theme$brand5.inapp) === null || _theme$brand5$inapp === void 0 ? void 0 : (_theme$brand5$inapp$t = _theme$brand5$inapp.toast) === null || _theme$brand5$inapp$t === void 0 ? void 0 : _theme$brand5$inapp$t.borderRadius) !== null && _theme$brand$inapp$to5 !== void 0 ? _theme$brand$inapp$to5 : "6px"
  }, theme === null || theme === void 0 ? void 0 : theme.progressBar)), _ref2;
};

exports.toastStyles = toastStyles;
var ToastStyled = (0, _styledComponents["default"])(_reactToastify.ToastContainer).withConfig({
  displayName: "styled__ToastStyled",
  componentId: "ycx4qo-0"
})(toastStyles);
exports.ToastStyled = ToastStyled;
