// @trycourier/react-inbox/dist/components/Message/styled.js
"use strict";

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.getIcon = exports.Icon = exports.UnreadIndicator = exports.TimeAgo = exports.ActionBlock = exports.TextBlock = exports.Title = exports.Contents = exports.Container = void 0;

var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));

var _react = _interopRequireWildcard(require("react"));

var _deepExtend = _interopRequireDefault(require("deep-extend"));

var _styledComponents = _interopRequireDefault(require("styled-components"));

function ownKeys(object, enumerableOnly) { var keys = Object.keys(object); if (Object.getOwnPropertySymbols) { var symbols = Object.getOwnPropertySymbols(object); if (enumerableOnly) symbols = symbols.filter(function (sym) { return Object.getOwnPropertyDescriptor(object, sym).enumerable; }); keys.push.apply(keys, symbols); } return keys; }

function _objectSpread(target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i] != null ? arguments[i] : {}; if (i % 2) { ownKeys(Object(source), true).forEach(function (key) { (0, _defineProperty2["default"])(target, key, source[key]); }); } else if (Object.getOwnPropertyDescriptors) { Object.defineProperties(target, Object.getOwnPropertyDescriptors(source)); } else { ownKeys(Object(source)).forEach(function (key) { Object.defineProperty(target, key, Object.getOwnPropertyDescriptor(source, key)); }); } } return target; }

var CourierSvg = function CourierSvg(props) {
  return /*#__PURE__*/_react["default"].createElement("svg", props, /*#__PURE__*/_react["default"].createElement("g", {
    fill: "#9121C2",
    fillRule: "nonzero"
  }, /*#__PURE__*/_react["default"].createElement("path", {
    d: "M24.838 10.662a.306.306 0 0 0 .074-.294C23.882 4.515 18.779.058 12.618.058 5.69.059.088 5.692.148 12.619.19 19.338 5.867 24.985 12.587 25c5.706.015 10.53-3.809 12.03-9.03.058-.176-.044-.367-.22-.44l-.516-.192c-2-.706-4.22-.456-6 .706-.808.53-1.411.97-1.411.97a6.284 6.284 0 0 1-3.56 1.089c-3.514 0-5.985-2.868-6.367-6.368l-.28-1.911a1.666 1.666 0 0 0-1.029-1.295l-.72-.279a.193.193 0 0 1-.089-.309c3.015-4.03 6.692-3.147 6.692-3.147a3.501 3.501 0 0 1 1.647.603c.661.456 1.161 1.103 1.485 1.838A7.34 7.34 0 0 0 21 11.647c.015 0 2.882.118 3.838-.985z"
  }), /*#__PURE__*/_react["default"].createElement("path", {
    d: "M7.809 7.647a.72.72 0 1 0 0-1.441.72.72 0 0 0 0 1.441zM7.97 11.456c.177 1.132 1.074 4.72 5.06 5.382.058.015.088-.073.029-.103-.927-.426-3.338-1.823-5-5.323-.015-.03-.103-.015-.088.044z"
  })));
};

CourierSvg.defaultProps = {
  width: "25",
  height: "25",
  xmlns: "http://www.w3.org/2000/svg"
};

var Container = _styledComponents["default"].div.withConfig({
  displayName: "styled__Container",
  componentId: "ri6fzz-0"
})(function (_ref) {
  var _theme$message$contai, _theme$message;

  var theme = _ref.theme;
  return (0, _deepExtend["default"])({
    display: "flex",
    position: "relative",
    padding: "10px 12px 10px 30px",
    backgroundColor: "#F9FAFB",
    alignItems: "center",
    marginTop: 6,
    borderRadius: 6,
    borderBottom: "1px solid rgba(203,213,224,.5)",
    "&.read": {
      backgroundColor: "#F7F6F9"
    },
    "&:not(.read):hover": {
      background: "#EDE4ED"
    }
  }, (_theme$message$contai = (_theme$message = theme.message) === null || _theme$message === void 0 ? void 0 : _theme$message.container) !== null && _theme$message$contai !== void 0 ? _theme$message$contai : {});
});

exports.Container = Container;

var Contents = _styledComponents["default"].div.withConfig({
  displayName: "styled__Contents",
  componentId: "ri6fzz-1"
})(function (_ref2) {
  var _theme$message2;

  var theme = _ref2.theme;
  return _objectSpread({
    marginRight: "auto",
    marginLeft: 15,
    textAlign: "left"
  }, (_theme$message2 = theme.message) === null || _theme$message2 === void 0 ? void 0 : _theme$message2.contents);
});

exports.Contents = Contents;

var Title = _styledComponents["default"].div.withConfig({
  displayName: "styled__Title",
  componentId: "ri6fzz-2"
})(function (_ref3) {
  var _theme$message3;

  var theme = _ref3.theme;
  return (0, _deepExtend["default"])({
    fontSize: "14px",
    fontStyle: "normal",
    fontWeight: "600",
    lineHeight: "19px",
    textAlign: "left",
    display: "-webkit-box",
    overflow: "hidden",
    textOverflow: "ellipsis",
    color: "#24324B"
  }, (_theme$message3 = theme.message) === null || _theme$message3 === void 0 ? void 0 : _theme$message3.title);
});

exports.Title = Title;

var TextBlock = _styledComponents["default"].div.withConfig({
  displayName: "styled__TextBlock",
  componentId: "ri6fzz-3"
})(function (_ref4) {
  var _theme$message4;

  var theme = _ref4.theme;
  return (0, _deepExtend["default"])({
    color: "#73819B",
    marginTop: "1px",
    wordBreak: "break-word",
    fontSize: "12px",
    fontStyle: "normal",
    fontWeight: "400",
    lineHeight: "16px",
    textAlign: "left",
    overflow: "hidden",
    textOverflow: "ellipsis"
  }, (_theme$message4 = theme.message) === null || _theme$message4 === void 0 ? void 0 : _theme$message4.textBlock);
});

exports.TextBlock = TextBlock;

var ActionBlock = _styledComponents["default"].div.withConfig({
  displayName: "styled__ActionBlock",
  componentId: "ri6fzz-4"
})(function (_ref5) {
  var _theme$brand$colors$p, _theme$brand, _theme$brand$colors, _theme$message5;

  var theme = _ref5.theme;
  return (0, _deepExtend["default"])({
    a: {
      display: "inline-block",
      cursor: "pointer",
      border: "none",
      fontSize: 12,
      color: "white",
      backgroundColor: (_theme$brand$colors$p = theme === null || theme === void 0 ? void 0 : (_theme$brand = theme.brand) === null || _theme$brand === void 0 ? void 0 : (_theme$brand$colors = _theme$brand.colors) === null || _theme$brand$colors === void 0 ? void 0 : _theme$brand$colors.primary) !== null && _theme$brand$colors$p !== void 0 ? _theme$brand$colors$p : "#9121C2",
      padding: "6px 15px",
      marginTop: 3,
      boxShadow: "0px 2px 8px rgba(0, 0, 0, 0.1)",
      textDecoration: "none",
      "&:active": {
        boxShadow: "none"
      },
      "&:hover": {
        color: "#73819B",
        background: "rgb(0 0 0 / 10%)"
      },
      borderRadius: 4
    }
  }, theme === null || theme === void 0 ? void 0 : (_theme$message5 = theme.message) === null || _theme$message5 === void 0 ? void 0 : _theme$message5.actionBlock);
});

exports.ActionBlock = ActionBlock;

var TimeAgo = _styledComponents["default"].div.withConfig({
  displayName: "styled__TimeAgo",
  componentId: "ri6fzz-5"
})(function (_ref6) {
  var _theme$message6;

  var theme = _ref6.theme;
  return (0, _deepExtend["default"])({
    color: "#aaa",
    fontSize: "10px",
    fontStyle: "normal",
    fontWeight: "400",
    lineHeight: "14px",
    whiteSpace: "nowrap",
    maxWidth: "65px",
    textOverflow: "ellipsis",
    overflow: "hidden"
  }, (_theme$message6 = theme.message) === null || _theme$message6 === void 0 ? void 0 : _theme$message6.timeAgo);
});

exports.TimeAgo = TimeAgo;

var iconStyles = function iconStyles(_ref7) {
  var _theme$brand$colors$p2, _theme$brand2, _theme$brand2$colors, _theme$message7;

  var theme = _ref7.theme;
  return (0, _deepExtend["default"])({
    height: 25,
    width: 25,
    flexShrink: "0",
    path: {
      fill: (_theme$brand$colors$p2 = theme === null || theme === void 0 ? void 0 : (_theme$brand2 = theme.brand) === null || _theme$brand2 === void 0 ? void 0 : (_theme$brand2$colors = _theme$brand2.colors) === null || _theme$brand2$colors === void 0 ? void 0 : _theme$brand2$colors.primary) !== null && _theme$brand$colors$p2 !== void 0 ? _theme$brand$colors$p2 : "#9121c2"
    }
  }, (_theme$message7 = theme.message) === null || _theme$message7 === void 0 ? void 0 : _theme$message7.icon);
};

var UnreadIndicator = _styledComponents["default"].div.withConfig({
  displayName: "styled__UnreadIndicator",
  componentId: "ri6fzz-6"
})(function (_ref8) {
  var _theme$brand$colors$p3, _theme$brand3, _theme$brand3$colors, _theme$message8;

  var theme = _ref8.theme;
  return (0, _deepExtend["default"])({
    height: 8,
    width: 8,
    backgroundColor: (_theme$brand$colors$p3 = theme === null || theme === void 0 ? void 0 : (_theme$brand3 = theme.brand) === null || _theme$brand3 === void 0 ? void 0 : (_theme$brand3$colors = _theme$brand3.colors) === null || _theme$brand3$colors === void 0 ? void 0 : _theme$brand3$colors.primary) !== null && _theme$brand$colors$p3 !== void 0 ? _theme$brand$colors$p3 : "#9121c2",
    borderRadius: "50%",
    position: "absolute",
    left: "13px"
  }, theme === null || theme === void 0 ? void 0 : (_theme$message8 = theme.message) === null || _theme$message8 === void 0 ? void 0 : _theme$message8.unreadIndicator);
});

exports.UnreadIndicator = UnreadIndicator;

var Icon = _styledComponents["default"].img.withConfig({
  displayName: "styled__Icon",
  componentId: "ri6fzz-7"
})(iconStyles);

exports.Icon = Icon;
var CourierIcon = (0, _styledComponents["default"])(CourierSvg).withConfig({
  displayName: "styled__CourierIcon",
  componentId: "ri6fzz-8"
})(iconStyles);

var getIcon = function getIcon(icon) {
  return (0, _react.useMemo)(function () {
    if (icon === false) {
      return;
    }

    if (icon && typeof icon === "string") {
      return /*#__PURE__*/_react["default"].createElement(Icon, {
        src: icon
      });
    }

    return /*#__PURE__*/_react["default"].createElement(CourierIcon, null);
  }, [icon]);
};

exports.getIcon = getIcon;
