// @trycourier/react-elements/dist/components/modal/modal.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Modal = void 0;

var _react = _interopRequireDefault(require("react"));

var _styledComponents = _interopRequireDefault(require("styled-components"));

var _title = require("../title");

var Modal = function Modal(_ref) {
  var isOpen = _ref.isOpen,
      onClose = _ref.onClose,
      title = _ref.title,
      children = _ref.children,
      className = _ref.className;
  return /*#__PURE__*/_react["default"].createElement("div", {
    className: className
  }, isOpen && /*#__PURE__*/_react["default"].createElement(Backdrop, {
    className: "modal-backdrop",
    onClick: onClose
  }), /*#__PURE__*/_react["default"].createElement(Dialog, {
    className: "modal",
    open: isOpen
  }, title && /*#__PURE__*/_react["default"].createElement(Header, {
    className: "modal-header"
  }, /*#__PURE__*/_react["default"].createElement(ModalTitle, {
    className: "modal-title",
    level: 3
  }, title)), /*#__PURE__*/_react["default"].createElement("div", {
    className: "modal-body"
  }, children)));
};

exports.Modal = Modal;

var Dialog = _styledComponents["default"].dialog.withConfig({
  displayName: "modal__Dialog",
  componentId: "s6yst4-0"
})(function () {
  return {
    background: "#24324B",
    border: "none",
    borderRadius: "0.25rem",
    overflow: "hidden",
    color: "white",
    padding: 0,
    width: 400,
    top: "50%",
    transform: "translateY(-50%)"
  };
});

var Backdrop = _styledComponents["default"].div.withConfig({
  displayName: "modal__Backdrop",
  componentId: "s6yst4-1"
})(function () {
  return {
    position: "fixed",
    content: "",
    top: 0,
    left: 0,
    right: 0,
    bottom: 0,
    zIndex: 0,
    backgroundColor: "rgba(0,0,0,0.5)"
  };
});

var Header = _styledComponents["default"].div.withConfig({
  displayName: "modal__Header",
  componentId: "s6yst4-2"
})(function () {
  return {
    display: "flex",
    justifyContent: "center",
    alignItems: "center",
    width: "100%",
    background: "rgb(88, 114, 162)",
    padding: "0.5rem",
    boxSizing: "border-box"
  };
});

var ModalTitle = (0, _styledComponents["default"])(_title.Title).withConfig({
  displayName: "modal__ModalTitle",
  componentId: "s6yst4-3"
})(function () {
  return {
    color: "white",
    margin: 0
  };
});
