// @trycourier/client-graphql/dist/inbox/index.js
"use strict";

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _client = require("../client");

var _count = require("./count");

var message = _interopRequireWildcard(require("./message"));

var messages = _interopRequireWildcard(require("./messages"));

var _markAllRead = require("./mark-all-read");

var _trackEvent = require("./track-event");

var _default = function _default(params) {
  var client = (0, _client.createCourierClient)(params, {
    apiUrl: "https://fxw3r7gdm9.execute-api.us-east-1.amazonaws.com/production/q"
  });
  return {
    getInboxCount: (0, _count.getInboxCount)(client),
    getMessage: message.getInboxMessage(client),
    getMessages: messages.getInboxMessages(client),
    markAllRead: (0, _markAllRead.markAllRead)(client),
    markArchive: (0, _trackEvent.trackEvent)(client)("archive"),
    markRead: (0, _trackEvent.trackEvent)(client)("read"),
    markUnread: (0, _trackEvent.trackEvent)(client)("unread")
  };
};

exports["default"] = _default;
