// @trycourier/react-provider/dist/index.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");

Object.defineProperty(exports, "__esModule", {
  value: true
});
var _exportNames = {
  registerReducer: true,
  CourierContext: true,
  CourierProvider: true
};
exports.CourierProvider = exports.CourierContext = exports.registerReducer = void 0;

var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));

var _slicedToArray2 = _interopRequireDefault(require("@babel/runtime/helpers/slicedToArray"));

var _toConsumableArray2 = _interopRequireDefault(require("@babel/runtime/helpers/toConsumableArray"));

var _react = _interopRequireWildcard(require("react"));

var _createReducer = _interopRequireDefault(require("react-use/lib/factory/createReducer"));

var _reducer = _interopRequireWildcard(require("./reducer"));

var _middleware2 = _interopRequireDefault(require("./middleware"));

var _useCourierActions = _interopRequireDefault(require("./hooks/use-courier-actions"));

var _useTransport = _interopRequireDefault(require("./hooks/use-transport"));

var _transports = require("./transports");

Object.keys(_transports).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _transports[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _transports[key];
    }
  });
});

var _hooks = require("./hooks");

Object.keys(_hooks).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _hooks[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _hooks[key];
    }
  });
});

var _global;

function ownKeys(object, enumerableOnly) { var keys = Object.keys(object); if (Object.getOwnPropertySymbols) { var symbols = Object.getOwnPropertySymbols(object); if (enumerableOnly) symbols = symbols.filter(function (sym) { return Object.getOwnPropertyDescriptor(object, sym).enumerable; }); keys.push.apply(keys, symbols); } return keys; }

function _objectSpread(target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i] != null ? arguments[i] : {}; if (i % 2) { ownKeys(Object(source), true).forEach(function (key) { (0, _defineProperty2["default"])(target, key, source[key]); }); } else if (Object.getOwnPropertyDescriptors) { Object.defineProperties(target, Object.getOwnPropertyDescriptors(source)); } else { ownKeys(Object(source)).forEach(function (key) { Object.defineProperty(target, key, Object.getOwnPropertyDescriptor(source, key)); }); } } return target; }

if (typeof window !== "undefined") {
  window.global = window; // @ts-ignore
  // eslint-disable-next-line @typescript-eslint/no-var-requires

  window.Buffer = window.Buffer || require("buffer").Buffer;
}

if (!((_global = global) !== null && _global !== void 0 && _global.localStorage)) {
  require("localstorage-polyfill");
}

var registerReducer = _reducer.registerReducer;
exports.registerReducer = registerReducer;

var CourierContext = /*#__PURE__*/_react["default"].createContext(undefined);

exports.CourierContext = CourierContext;

var CourierProvider = function CourierProvider(_ref) {
  var apiUrl = _ref.apiUrl,
      brand = _ref.brand,
      brandId = _ref.brandId,
      children = _ref.children,
      clientKey = _ref.clientKey,
      _ref$middleware = _ref.middleware,
      _middleware = _ref$middleware === void 0 ? [] : _ref$middleware,
      onMessage = _ref.onMessage,
      _transport = _ref.transport,
      userId = _ref.userId,
      userSignature = _ref.userSignature,
      wsOptions = _ref.wsOptions;

  var middleware = [].concat((0, _toConsumableArray2["default"])(_middleware), (0, _toConsumableArray2["default"])(_middleware2["default"]));
  var useReducer = (0, _react.useCallback)(_createReducer["default"].apply(void 0, (0, _toConsumableArray2["default"])(middleware)), [_middleware]);
  var transport = (0, _useTransport["default"])({
    transport: _transport,
    userSignature: userSignature,
    clientKey: clientKey,
    wsOptions: wsOptions
  });

  var _useReducer = useReducer(_reducer["default"], {
    apiUrl: apiUrl,
    brand: brand,
    brandId: brandId,
    clientKey: clientKey,
    transport: transport,
    userId: userId,
    userSignature: userSignature,
    middleware: middleware
  }),
      _useReducer2 = (0, _slicedToArray2["default"])(_useReducer, 2),
      state = _useReducer2[0],
      dispatch = _useReducer2[1];

  var actions = (0, _useCourierActions["default"])(state, dispatch);
  (0, _react.useEffect)(function () {
    if (_transport) {
      // this means the transport was passed in and we shouldn't subscribe
      return;
    }

    if (!transport || !userId) {
      return;
    }

    var courierTransport = transport;
    courierTransport.subscribe(userId);

    if (onMessage) {
      courierTransport.intercept(onMessage);
    }

    courierTransport.listen({
      id: "deliver-tracking",
      listener: function listener(courierEvent) {
        var _courierEvent$data, _courierData$tracking, _courierData$tracking2;

        var courierData = courierEvent === null || courierEvent === void 0 ? void 0 : (_courierEvent$data = courierEvent.data) === null || _courierEvent$data === void 0 ? void 0 : _courierEvent$data.data;

        if (!(courierData !== null && courierData !== void 0 && (_courierData$tracking = courierData.trackingIds) !== null && _courierData$tracking !== void 0 && _courierData$tracking.deliverTrackingId)) {
          return;
        }

        actions.createTrackEvent(courierData === null || courierData === void 0 ? void 0 : (_courierData$tracking2 = courierData.trackingIds) === null || _courierData$tracking2 === void 0 ? void 0 : _courierData$tracking2.deliverTrackingId);
      }
    });
    return function () {
      courierTransport.unsubscribe(userId);
    };
  }, [actions, transport, userId]);
  (0, _react.useEffect)(function () {
    if (!_transport && (!clientKey || !userId)) {
      return;
    }

    actions.init({
      apiUrl: apiUrl,
      brandId: brandId,
      clientKey: clientKey,
      transport: transport,
      userId: userId,
      userSignature: userSignature
    });
  }, [actions, apiUrl, brandId, clientKey, transport, userId, userSignature]);
  (0, _react.useEffect)(function () {
    if (!state.brand || !clientKey || !userId) {
      return;
    }

    localStorage.setItem("".concat(clientKey, "/").concat(userId, "/provider"), JSON.stringify({
      brand: state.brand
    }));
  }, [state.brand, clientKey, userId]);
  (0, _react.useEffect)(function () {
    if (!clientKey || !userId) {
      return;
    }

    var localStorageState = localStorage.getItem("".concat(clientKey, "/").concat(userId, "/provider"));

    if (localStorageState) {
      try {
        var _JSON$parse = JSON.parse(localStorageState),
            _brand = _JSON$parse.brand;

        actions.setBrand(_brand);
      } catch (ex) {
        console.log("error", ex);
      }
    }
  }, [clientKey, userId]);
  return /*#__PURE__*/_react["default"].createElement(CourierContext.Provider, {
    value: _objectSpread(_objectSpread(_objectSpread({}, state), actions), {}, {
      dispatch: dispatch
    })
  }, children);
};

exports.CourierProvider = CourierProvider;
