// @trycourier/react-hooks/dist/inbox/actions/fetch-message-lists.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.fetchMessageListsDone = exports.INBOX_FETCH_MESSAGE_LISTS_DONE = exports.fetchMessageListsError = exports.INBOX_FETCH_MESSAGE_LISTS_ERROR = exports.fetchMessageListsPending = exports.INBOX_FETCH_MESSAGE_LISTS_PENDING = exports.INBOX_FETCH_MESSAGE_LISTS = void 0;
var INBOX_FETCH_MESSAGE_LISTS = "inbox/FETCH_MESSAGE_LISTS";
exports.INBOX_FETCH_MESSAGE_LISTS = INBOX_FETCH_MESSAGE_LISTS;
var INBOX_FETCH_MESSAGE_LISTS_PENDING = "inbox/FETCH_MESSAGE_LISTS/PENDING";
exports.INBOX_FETCH_MESSAGE_LISTS_PENDING = INBOX_FETCH_MESSAGE_LISTS_PENDING;

var fetchMessageListsPending = function fetchMessageListsPending() {
  return {
    type: INBOX_FETCH_MESSAGE_LISTS_PENDING
  };
};

exports.fetchMessageListsPending = fetchMessageListsPending;
var INBOX_FETCH_MESSAGE_LISTS_ERROR = "inbox/FETCH_MESSAGE_LISTS/ERROR";
exports.INBOX_FETCH_MESSAGE_LISTS_ERROR = INBOX_FETCH_MESSAGE_LISTS_ERROR;

var fetchMessageListsError = function fetchMessageListsError() {
  return {
    type: INBOX_FETCH_MESSAGE_LISTS_ERROR
  };
};

exports.fetchMessageListsError = fetchMessageListsError;
var INBOX_FETCH_MESSAGE_LISTS_DONE = "inbox/FETCH_MESSAGE_LISTS/DONE";
exports.INBOX_FETCH_MESSAGE_LISTS_DONE = INBOX_FETCH_MESSAGE_LISTS_DONE;

var fetchMessageListsDone = function fetchMessageListsDone(payload) {
  return {
    type: INBOX_FETCH_MESSAGE_LISTS_DONE,
    payload: payload
  };
};

exports.fetchMessageListsDone = fetchMessageListsDone;
