// @trycourier/client-graphql/dist/client.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.createCourierClient = void 0;

var _urql = require("urql");

var createCourierClient = function createCourierClient(params, defaults) {
  if ("client" in params) {
    return params.client;
  }

  var headers;

  if ("authorization" in params) {
    headers = {
      authorization: "Bearer ".concat(params.authorization)
    };
  } else {
    var _ref = params,
        clientKey = _ref.clientKey,
        userId = _ref.userId,
        userSignature = _ref.userSignature;
    headers = {
      "x-courier-client-key": clientKey,
      "x-courier-user-id": userId
    };

    if (userSignature) {
      headers["x-courier-user-signature"] = userSignature;
    }

    if (!clientKey || !userId) {
      return;
    }
  }

  return (0, _urql.createClient)({
    url: params.apiUrl || (defaults === null || defaults === void 0 ? void 0 : defaults.apiUrl) || "".concat(undefined || "https://api.courier.com", "/client/q"),
    requestPolicy: "network-only",
    fetchOptions: function fetchOptions() {
      return {
        headers: headers
      };
    }
  });
};

exports.createCourierClient = createCourierClient;
