// @trycourier/client-graphql/dist/preferences.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = exports.updateRecipientPreferences = exports.getRecipientPreferences = void 0;

var _regenerator = _interopRequireDefault(require("@babel/runtime/regenerator"));

var _asyncToGenerator2 = _interopRequireDefault(require("@babel/runtime/helpers/asyncToGenerator"));

var _client = require("./client");

var RECIPIENT_PREFERENCES = "\n  query GetRecipientPreferences {\n    recipientPreferences {\n      nodes {\n        templateId\n        status\n        hasCustomRouting\n        routingPreferences\n      }\n    }\n  }\n";

var getRecipientPreferences = function getRecipientPreferences(client) {
  return /*#__PURE__*/(0, _asyncToGenerator2["default"])( /*#__PURE__*/_regenerator["default"].mark(function _callee() {
    var _results$data;

    var results;
    return _regenerator["default"].wrap(function _callee$(_context) {
      while (1) {
        switch (_context.prev = _context.next) {
          case 0:
            if (client) {
              _context.next = 2;
              break;
            }

            return _context.abrupt("return");

          case 2:
            _context.next = 4;
            return client.query(RECIPIENT_PREFERENCES).toPromise();

          case 4:
            results = _context.sent;
            return _context.abrupt("return", (_results$data = results.data) === null || _results$data === void 0 ? void 0 : _results$data.recipientPreferences.nodes);

          case 6:
          case "end":
            return _context.stop();
        }
      }
    }, _callee);
  }));
};

exports.getRecipientPreferences = getRecipientPreferences;
var UPDATE_RECIPIENT_PREFERENCES = "\n  mutation UpdateRecipientPreferences($id: String!, $preferences: PreferencesInput!) {\n    updatePreferences(templateId: $id preferences: $preferences)\n  }\n";

var updateRecipientPreferences = function updateRecipientPreferences(client) {
  return /*#__PURE__*/function () {
    var _ref2 = (0, _asyncToGenerator2["default"])( /*#__PURE__*/_regenerator["default"].mark(function _callee2(payload) {
      return _regenerator["default"].wrap(function _callee2$(_context2) {
        while (1) {
          switch (_context2.prev = _context2.next) {
            case 0:
              if (client) {
                _context2.next = 2;
                break;
              }

              return _context2.abrupt("return", Promise.resolve());

            case 2:
              _context2.next = 4;
              return client.mutation(UPDATE_RECIPIENT_PREFERENCES, {
                id: payload.templateId,
                preferences: {
                  status: payload.status,
                  hasCustomRouting: payload.hasCustomRouting,
                  routingPreferences: payload.routingPreferences
                }
              }).toPromise();

            case 4:
              return _context2.abrupt("return", payload);

            case 5:
            case "end":
              return _context2.stop();
          }
        }
      }, _callee2);
    }));

    return function (_x) {
      return _ref2.apply(this, arguments);
    };
  }();
};

exports.updateRecipientPreferences = updateRecipientPreferences;

var _default = function _default(params) {
  var client = (0, _client.createCourierClient)(params);
  return {
    getRecipientPreferences: getRecipientPreferences(client),
    updateRecipientPreferences: updateRecipientPreferences(client)
  };
};

exports["default"] = _default;
