// @trycourier/react-elements/dist/components/buttons/styles.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.genButtonStyles = void 0;

var genButtonStyles = function genButtonStyles(_ref) {
  var _ref$size = _ref.size,
      size = _ref$size === void 0 ? "sm" : _ref$size,
      color = _ref.color,
      textColor = _ref.textColor;

  var _getButtonSize = getButtonSize(size),
      height = _getButtonSize.height,
      fontSize = _getButtonSize.fontSize;

  return {
    display: "inline-flex",
    cursor: "pointer",
    border: "none",
    fontSize: fontSize,
    height: height,
    width: "auto",
    color: textColor !== null && textColor !== void 0 ? textColor : "white",
    backgroundColor: color !== null && color !== void 0 ? color : "#9121C2",
    padding: "4px 15px",
    textDecoration: "none",
    boxSizing: "border-box",
    flexDirection: "row",
    alignItems: "center",
    justifyContent: "center",
    lineHeight: "1",
    "&:hover": {
      color: "#73819B",
      background: "rgb(0 0 0 / 10%)"
    },
    "&:disabled": {
      color: "#73819B",
      background: "rgb(0 0 0 / 10%)"
    },
    borderRadius: 4
  };
};

exports.genButtonStyles = genButtonStyles;

function getButtonSize(size) {
  switch (size) {
    case "xs":
      return {
        height: 30,
        fontSize: 12
      };

    case "sm":
      return {
        height: 34,
        fontSize: 14
      };

    case "md":
      return {
        height: 38,
        fontSize: 16
      };

    case "lg":
      return {
        height: 42,
        fontSize: 18
      };

    default:
      return {
        height: 30,
        fontSize: 12
      };
  }
}
