// @trycourier/react-toast/dist/hooks/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.useListenForTransportEvent = exports.useToast = void 0;

var _reactProvider = require("@trycourier/react-provider");

var _react = require("react");

var useToast = function useToast() {
  var _toast$config;

  var _useCourier = (0, _reactProvider.useCourier)(),
      toast = _useCourier.toast,
      clientKey = _useCourier.clientKey;

  var toastCaller = toast !== null && toast !== void 0 && toast.toast ? toast.toast : function () {};
  return [toastCaller, {
    config: (_toast$config = toast === null || toast === void 0 ? void 0 : toast.config) !== null && _toast$config !== void 0 ? _toast$config : {},
    clientKey: clientKey
  }];
};

exports.useToast = useToast;

var useListenForTransportEvent = function useListenForTransportEvent(clientKey, transport, handleToast) {
  (0, _react.useEffect)(function () {
    if (!transport) {
      return;
    }

    transport.listen({
      id: "toast-listener",
      listener: function listener(courierEvent) {
        handleToast(courierEvent === null || courierEvent === void 0 ? void 0 : courierEvent.data);
      }
    });
  }, [clientKey, handleToast, transport]);
};

exports.useListenForTransportEvent = useListenForTransportEvent;
