// @trycourier/react-elements/dist/components/pretty-date.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.PrettyDate = void 0;

var _react = _interopRequireDefault(require("react"));

var _styledComponents = _interopRequireDefault(require("styled-components"));

var PrettyDate = function PrettyDate(opts) {
  var _opts$dateFormat;

  return /*#__PURE__*/_react["default"].createElement(PrettyDateContainer, null, opts.prefix, new Date().toLocaleDateString(opts.locale, (_opts$dateFormat = opts.dateFormat) !== null && _opts$dateFormat !== void 0 ? _opts$dateFormat : {
    month: "long",
    day: "numeric",
    year: "numeric"
  }));
};

exports.PrettyDate = PrettyDate;

var PrettyDateContainer = _styledComponents["default"].p.withConfig({
  displayName: "pretty-date__PrettyDateContainer",
  componentId: "sc-1vk9gg6-0"
})(function (_ref) {
  var color = _ref.color;
  return {
    color: color !== null && color !== void 0 ? color : "rgba(115, 129, 155, 0.5);",
    fontFamily: "'Nunito Sans', sans-serif",
    fontWeight: 400,
    fontSize: "12px",
    lineHeight: "16px",
    letterSpacing: "-0.02em",
    display: "inline-block"
  };
});
