// @trycourier/react-elements/dist/components/title.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Title = void 0;

var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));

var _objectWithoutProperties2 = _interopRequireDefault(require("@babel/runtime/helpers/objectWithoutProperties"));

var _react = _interopRequireDefault(require("react"));

var _styledComponents = _interopRequireDefault(require("styled-components"));

var getTitleStyles = function getTitleStyles(_ref) {
  var size = _ref.size,
      weight = _ref.weight,
      color = _ref.color;
  return {
    fontSize: size !== null && size !== void 0 ? size : "16px",
    fontStyle: "normal",
    fontWeight: weight !== null && weight !== void 0 ? weight : 800,
    fontFamily: "'Nunito Sans', sans-serif",
    lineHeight: "19px",
    textAlign: "left",
    display: "-webkit-box",
    overflow: "hidden",
    textOverflow: "ellipsis",
    color: color !== null && color !== void 0 ? color : "#24324B"
  };
};

var headings = {
  H1: _styledComponents["default"].h1.withConfig({
    displayName: "title__H1",
    componentId: "sc-19e2nvs-0"
  })(getTitleStyles),
  H2: _styledComponents["default"].h2.withConfig({
    displayName: "title__H2",
    componentId: "sc-19e2nvs-1"
  })(getTitleStyles),
  H3: _styledComponents["default"].h3.withConfig({
    displayName: "title__H3",
    componentId: "sc-19e2nvs-2"
  })(getTitleStyles),
  H4: _styledComponents["default"].h4.withConfig({
    displayName: "title__H4",
    componentId: "sc-19e2nvs-3"
  })(getTitleStyles),
  H5: _styledComponents["default"].h5.withConfig({
    displayName: "title__H5",
    componentId: "sc-19e2nvs-4"
  })(getTitleStyles)
};

var Title = function Title(opts) {
  var _headings$;

  var _opts$level = opts.level,
      level = _opts$level === void 0 ? 2 : _opts$level,
      rest = (0, _objectWithoutProperties2["default"])(opts, ["level"]);
  var Heading = (_headings$ = headings["H".concat(level)]) !== null && _headings$ !== void 0 ? _headings$ : headings.H2;
  var index = Math.min(Math.max(level - 1, 0), 4);
  var weight = ["900", "800", "700", "500", "300"][index];
  var size = ["18px", "16px", "14px", "12px", "10px"][index];
  return /*#__PURE__*/_react["default"].createElement(Heading, (0, _extends2["default"])({
    weight: weight,
    size: size
  }, rest));
};

exports.Title = Title;
