// @trycourier/react-preferences/dist/components/PreferenceList.js
"use strict";

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.PreferenceList = exports.StyledList = void 0;

var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));

var _react = _interopRequireWildcard(require("react"));

var _reactHooks = require("@trycourier/react-hooks");

var _reactProvider = require("@trycourier/react-provider");

var _PreferenceTemplate = require("./PreferenceTemplate");

var _styledComponents = _interopRequireWildcard(require("styled-components"));

function ownKeys(object, enumerableOnly) { var keys = Object.keys(object); if (Object.getOwnPropertySymbols) { var symbols = Object.getOwnPropertySymbols(object); if (enumerableOnly) symbols = symbols.filter(function (sym) { return Object.getOwnPropertyDescriptor(object, sym).enumerable; }); keys.push.apply(keys, symbols); } return keys; }

function _objectSpread(target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i] != null ? arguments[i] : {}; if (i % 2) { ownKeys(Object(source), true).forEach(function (key) { (0, _defineProperty2["default"])(target, key, source[key]); }); } else if (Object.getOwnPropertyDescriptors) { Object.defineProperties(target, Object.getOwnPropertyDescriptors(source)); } else { ownKeys(Object(source)).forEach(function (key) { Object.defineProperty(target, key, Object.getOwnPropertyDescriptor(source, key)); }); } } return target; }

var StyledList = _styledComponents["default"].div.withConfig({
  displayName: "PreferenceList__StyledList",
  componentId: "sc-14ehsz3-0"
})(["overflow:scroll;display:flex;height:433px;flex-direction:column;border-top:1px solid rgba(203,213,224,0.5);background:rgba(255,255,255,0.2);"]);

exports.StyledList = StyledList;

var PreferenceList = function PreferenceList(props) {
  var _brand$preferenceTemp, _brand$preferenceTemp2;

  var _useCourier = (0, _reactProvider.useCourier)(),
      brand = _useCourier.brand;

  var preferences = (0, _reactHooks.usePreferences)();
  (0, _react.useEffect)(function () {
    preferences.fetchRecipientPreferences();
  }, []);
  return /*#__PURE__*/_react["default"].createElement(_react["default"].Fragment, null, /*#__PURE__*/_react["default"].createElement(_styledComponents.ThemeProvider, {
    theme: _objectSpread(_objectSpread({}, props.theme), {}, {
      brand: brand
    })
  }, /*#__PURE__*/_react["default"].createElement(StyledList, null, preferences !== null && preferences !== void 0 && preferences.isLoading || !(brand !== null && brand !== void 0 && (_brand$preferenceTemp = brand.preferenceTemplates) !== null && _brand$preferenceTemp !== void 0 && _brand$preferenceTemp.length) ? /*#__PURE__*/_react["default"].createElement(_react["default"].Fragment, null) : brand === null || brand === void 0 ? void 0 : (_brand$preferenceTemp2 = brand.preferenceTemplates) === null || _brand$preferenceTemp2 === void 0 ? void 0 : _brand$preferenceTemp2.map(function (template) {
    var _preferences$recipien;

    return /*#__PURE__*/_react["default"].createElement(_PreferenceTemplate.PreferenceTemplate, {
      key: template.templateId,
      preferenceTemplate: template,
      recipientPreference: preferences === null || preferences === void 0 ? void 0 : (_preferences$recipien = preferences.recipientPreferences) === null || _preferences$recipien === void 0 ? void 0 : _preferences$recipien.find(function (preference) {
        return preference.templateId === template.templateId;
      })
    });
  }))));
};

exports.PreferenceList = PreferenceList;
